llvm/lib/Transforms/Vectorize/LoadStoreVectorizer.cpp

```diff
@@ -626,26 +626,35 @@ std::vector<Chain> Vectorizer::splitChainByContiguity(Chain &C) {
   std::vector<Chain> Ret;
   Ret.push_back({C.front()});
 
+  unsigned ElemBytes = DL.getTypeStoreSize(getChainElemTy(C));
```

Comment:
I think this usage of getTypeStoreSize here, specifically with getChainElemTy, might be wrong, because while the chain cannot contain sub-byte-sized elements (as the previous code asserted) they _can_ contain elements that are _themselves_ vectors of i1s. Calling getTypeStoreSize on getChainElemTy that returns i1 might be wrong in those cases. See /test/Transforms/LoadStoreVectorizer/NVPTX/vectorize_i1.ll as an example of i1 types the vectorizer handles.

To be safe, can you make a unit test with i1 vector elements?